graveler/graveler.go

```diff
@@ -864,60 +863,44 @@ func newStagingToken(repositoryID RepositoryID, branchID BranchID) StagingToken
 	return StagingToken(v)
 }
 
-func (g *graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, branchID BranchID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
-	res, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, branchID, func() (interface{}, error) {
-		repo, err := g.RefManager.GetRepository(ctx, repositoryID)
-		if err != nil {
-			return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
-		}
-		branch, err := g.RefManager.GetBranch(ctx, repositoryID, branchID)
-		if err != nil {
-			return "", fmt.Errorf("get branch %s: %w", branchID, err)
-		}
-		if empty, err := g.stagingEmpty(ctx, branch); err != nil {
-			return "", err
-		} else if !empty {
-			return "", ErrDirtyBranch
-		}
+func (g *graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
```

Comment:
Just fyi we would probably need to update this method to accept Commit (struct) in order to control the time and parents (more than one) to migrate.